### PR TITLE
UBL - minor changes/suggestion 2

### PR DESCRIPTION
**Pull request by Bob-the-Kuhn**
Changed the P2 edit function to be similar to the P4 edit function.  This required adding functions similar to ftostr43sign & lcd_mesh_edit.  It's now obvious where the variable being modified is on the screen.

Changed P4 so that unprobed points are not tuned.  Currently fine tuning an unprobed point will result in garbage in the editor and a huge number being written to the point.

----

Sometimes when the editor screen first comes up all you see is a blank screen.  It's not until you turn the encoder that the text pops up.  Don't know if that's semi-usual or if there's a fix for that.

### CODE REVIEW

**Comment by Roxy-3D** [comment]
> Changed P4 so that unprobed points are not tuned. Currently fine tuning an unprobed point will result in garbage in the editor and a huge number being written to the point.

Thanks!

> Sometimes when the editor screen first comes up all you see is a blank screen. It's not until you turn the encoder that the text pops up. Don't know if that's semi-usual or if there's a fix for that.

I have not seen that before.  Are you on a 20x4 LCD Panel or a Graphics Panel?

**Comment by Bob-the-Kuhn** [comment]
I'm running a Viki2 which is a graphics display.

File: Marlin/ultralcd.cpp [hunk 1]
```diff
@@ -924,6 +945,11 @@ void kill_screen(const char* lcd_msg) {
       return mesh_edit_value;
     }
 
+    float lcd_manual_probe() {
+      lcd_goto_screen(_lcd_mesh_edit_NOP);
```
Comment (thinkyhead) [inline Marlin/ultralcd.cpp:949]: @Roxy-3D How does other code manage interaction with the LCD without requiring `_lcd_mesh_edit_NOP` and other hacks?

Take a look at the methods used by `gcode_M600` to take over the controller for display and feedback purposes. Take a look at how `LCD_BED_LEVELING` does its UI as an LCD-initiated process.

For U.B.L. I'd recommend following the example of `gcode_M600`. In the process of fixing the LCD interaction code, I'm sure new opportunities for further unification and improvement of our UI methods will jump out.

The main thing is, when we see we're using hacks, it's time to take a step back.

**Comment by thinkyhead** [comment]
> Sometimes when the editor screen first comes up all you see is a blank screen.

See `gcode_M600` for a function that interacts with the UI in the recommended fashion.

**Comment by Bob-the-Kuhn** [comment]
All items in thhis PR are either not being done or are being done on PR #6501 .

**Pull request closed without merging**